File: app/src/main/java/com/pnuema/bible/ui/utils/DialogUtils.java
package com.pnuema.bible.ui.utils;

import androidx.fragment.app.FragmentActivity;

import com.pnuema.bible.ui.dialogs.BCVDialog;
import com.pnuema.bible.ui.dialogs.NotifySelectionCompleted;
import com.pnuema.bible.ui.dialogs.NotifyVersionSelectionCompleted;
import com.pnuema.bible.ui.dialogs.VersionSelectionDialog;

/**
 * Dialog utilities to generate and display dialogs.
 */
public final class DialogUtils {
    private DialogUtils() {
    }

    public static void showBookChapterVersePicker(final FragmentActivity activity, final BCVDialog.BCV bcv, final NotifySelectionCompleted listener) {
        BCVDialog.instantiate(bcv, listener).show(activity.getSupportFragmentManager(), BCVDialog.class.getSimpleName());
    }

    public static void showVersionsPicker(final FragmentActivity activity, final NotifyVersionSelectionCompleted listener) {
        VersionSelectionDialog.instantiate(listener).show(activity.getSupportFragmentManager(), VersionSelectionDialog.class.getSimpleName());
    }
}

File: app/src/main/kotlin/com/pnuema/bible/android/ui/utils/DialogUtils.kt
package com.pnuema.bible.android.ui.utils

import androidx.fragment.app.FragmentManager
import com.pnuema.bible.android.ui.bookchapterverse.BCVDialog
import com.pnuema.bible.android.ui.bookchapterverse.NotifySelectionCompleted

/**
 * Dialog utilities to generate and display dialogs.
 */
object DialogUtils {

    fun showBookChapterVersePicker(fragmentManager: FragmentManager, bcv: BCVDialog.BCV, listener: NotifySelectionCompleted) {
        /*fragmentManager.commit {
            setCustomAnimations(androidx.appcompat.R.anim.abc_slide_in_top, 0, 0, 0)
            addToBackStack(BCVDialog::class.java.simpleName)
            replace(R.id.read_fragment, BCVDialog.instantiate(bcv, listener), BCVDialog::class.java.simpleName)
        }*/
    }
}